2/contrib/jenkins/kube-slave-common.sh

```diff
@@ -31,6 +31,10 @@ export JNLP_PORT=${JNLP_PORT:-50000}
 export NODEJS_SLAVE=${NODEJS_SLAVE_IMAGE:-image-registry.openshift-image-registry.svc:5000/openshift/jenkins-agent-nodejs:latest}
 export MAVEN_SLAVE=${MAVEN_SLAVE_IMAGE:-image-registry.openshift-image-registry.svc:5000/openshift/jenkins-agent-maven:latest}
 
+export AGENT_BASE=${AGENT_BASE_IMAGE:-image-registry.openshift-image-registry.svc:5000/openshift/jenkins-agent-base:latest}
```

Comment:
can we rename that into , respectively:

```
BASE_AGENT
MAVEN_AGENT
NODEJS_AGENT
```
As we haven't done the inclusive renaming yet, using these new agents names would be an opportunity to do so.
At the same time, the use of `SIDECAR` is more about an internal implementation detail and does not necessarily reflects what it does.

Reply:
I'll think about it, and we should probably discuss as a team.  Not that my current names could use some work, but at first blush I'm not a fan of those 3 names.

Also, I think sidecar is more than an implementation detail.  IMO it is a "concept" that is highlighted / discussed in this part of the jenkins ecosystem.  So using sidecar is both descriptive and distinguishing at the "concept" level, not the implementation detail level.

Reply:
@adambkaplan what is your opinion on the naming ? ^^

Reply:
@gabemontero I don't see any reference to the `sidecar` term in the kubernetes plugin documentation or code. There is the concept on "multiple containers podTemplates", but that would not help us to find a name.
At the end of the day, it is not important technically.

Reply:
I think `sidecar` is pretty synonymous with "muliple conatiner pods" in our domain here.  And it is a nice/concise way of referring to the "multiple container pods" we are talking about, so it is good IMO for names like this.

All that said, I'm all for getting quorum and going with what ever the consensus is.

@adambkaplan WDYT?

Reply:
oh and forgot ....MAVEN_AGENT and NODEJS_AGENT should be reserved as better language versions of the existing names we have here for the overrides for the existing two pod templates with only 1 container (though I'd prefer to handle the required deprecation etc. of those existing names under a different work item i.e. we cannot just replace them).

Reply:
At risk of turning this into a total bikeshed, might "ambassador" strike the right middle ground here [1]? Sidecar IMO is a bit of a loose term for multi-container pods, whereas "Ambassador" is a bit more formalized. Can we think of the JNLP agent as the "Jenkins ambassador" to the main container that runs the build (or other action in the Jenkins pipeline)?

Sidecar is perhaps a more common term, though. It may not be referenced directly in the client plugin docs, but the term "sidecar container" is fairly common in the Kubernetes community, especially with the growth of Istio and other service mesh platforms.

[1] https://docs.microsoft.com/en-us/azure/architecture/patterns/ambassador